src/Metrics.jl

```diff
@@ -80,11 +80,32 @@ end
 
 Surface normal integral of field `p` over the `body`.
 """
-function ∮nds(p::AbstractArray{T,N},df::AbstractArray{T},body::AbstractBody,t=0) where {T,N}
-    @loop df[I,:] .= p[I]*nds(body,loc(0,I,T),t) over I ∈ inside(p)
-    [sum(@inbounds(df[inside(p),i])) for i ∈ 1:N] |> Array
+∮nds(flow::Flow,body::AbstractBody) = ∮nds(flow.p,body,time(flow))
+∮nds(p::AbstractArray{T,N},body::AbstractBody,t=0) where {T,N} = sum(inside(p)) do I
+    d,n,_ = measure_fast(body,WaterLily.loc(0,I),t)
+    p[I]*n*WaterLily.kern(d)
 end
-@inline function nds(body::AbstractBody,x,t)
-    d,n,_ = measure(body,x,t)
-    n*WaterLily.kern(clamp(d,-1,1))
+# viscous stress tensor
+∇²u(I::CartesianIndex{2},u) = @SMatrix [∂(i,j,I,u)+∂(j,i,I,u) for i ∈ 1:2, j ∈ 1:2]
+∇²u(I::CartesianIndex{3},u) = @SMatrix [∂(i,j,I,u)+∂(j,i,I,u) for i ∈ 1:3, j ∈ 1:3]
+"""
+   ∮τnds(u,body::AbstractBody,t=0)
+
+Compute the viscous force on a immersed body. 
+"""
+∮τnds(flow::Flow,body::AbstractBody) = ∮τnds(flow.u,body,time(flow))
+∮τnds(u::AbstractArray{T,N},body::AbstractBody,t=0) where {T,N} = sum(inside_u(u)) do I
```

Comment:
```suggestion
∮τnds(u::AbstractArray,body::AbstractBody,t=0) = sum(inside_u(u)) do I
```

Reply:
It's not `inside_u(u)` but `inside_u(size(u))` right?

Reply:
No, that's why I had to write another function 'inside_u(u)` that points to the inside of the array, without the components indices, which would equate to `inside_u(size(u))[1:end-1]`. I just think it's cleaner that way.